res/controllers/Denon-MC7000-scripts.js

```diff
@@ -573,34 +610,154 @@ MC7000.PadButtons = function(channel, control, value, status, group) {
             midi.sendShortMsg(0x94 + deckOffset, control, MC7000.padColor.sliceron);
         }
     } else if (MC7000.PADModeSampler[deckOffset]) {
+        var samplerOffset = 0;
+        var samplerOffsetJ = 0;
         for (i = 1; i <= 8; i++) {
+            if (MC7000.SamplerQty === 16) {
+                if (deckOffset >= 2) {
+                    deckOffset = deckOffset - 2;
+                }
+            }
+            samplerOffset = deckOffset * 8 + i;
```

Comment:
```suggestion
            deckOffset = numberOffset * 8;    
```
This would be a real deck offset. And than you could use `deckOffset + samplerIdx` in the following code

Reply:
I am not sure if I understand this correctly, numberOffset is a new variable declared nowhere. Within the code modification from above I assume that now it should be something like 
deckIndex = deckNumber  - 1;
deckOffset = deckIndex * 8;
and
samplerOffset = deckOffset + samplerIdx;

So that's what I did.